### PR TITLE
student instructions for the team troubleshooting project

### PR DESCRIPTION
**Pull request by sciclic**
A draft document for providing instructions and details to students about the team troubleshooting project, still needs work but it's a start! I look forward to your feedback and I can change it accordingly. 

The way I was thinking about the grading:

+ Each milestone is 30 points total (15 from the actual deliverable, which has 30 "errors" to correct, 0.5 points per error + 15 from the pull request) x 3 milestones = 90 points

+ Repo and evaluation are 10 points

= 100 points, and then it's easier to translate to the 35% of the total course grade which is what the project is worth.

Let me know what you think.

Thank you,
Icíar

### CODE REVIEW

**Comment by sciclic** [comment]
I also started working on the first milestone "deliverable" for this project and intended to have it in a separate pull request but I see now that I committed it to this one, sorry! There are two documents: one is the actual deliverable for students, and the other are the solutions, but I just started working on it last night so there's still a lot to do. This is the commit https://github.com/UBC-STAT/stat545.stat.ubc.ca/commit/3349ad2c67d3ffc0b4a5a7867d5e79adb177435f, please let me know if you have any comments on what I've done so far and thanks again!

File: 2020-materials/team-troubleshooting/student-instructions.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
++  The team troubleshooting project accounts for 35% of the total course grade.
+
++ Each milestone is graded as a group grade.
+  + Each deliverable is graded out of 15 points (each error is 0.5 points).
+  + Pull requests are graded out of 15 points, based on:
+    + A good title and description. A good title is self-explanatory, and a good description clearly states *what* was changed in the pull request and *why*.
+    + Productive discussion. Both students engaged in discussion in the pull request, and made changes to the document accordingly.
+
++ At the end of the project, each member will submit a quick evaluation of the other team members, and themselves. This evaluation is worth 10 points.
```
Comment (vincenzocoia) [inline 2020-materials/team-troubleshooting/student-instructions.md:40]: > This evaluation is worth 10 points.

To clarify -- are we saying that *completing* this document is worth 10 points? And that, if there are problems with your teamwork, you will be assigned an overall multiplier to your grade (like 0.8*grade or something)?

File: 2020-materials/team-troubleshooting/TB1.Rmd [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  + arrange
+  + summarize
+  + join 
+  
+## Instructions
+
+There are **30 errors** in this Rmd. Will you find them all?
+
++ *20 errors are marked* by ## ERROR HERE ## in the code chunk in which they are located.
++ *10 errors are hiding.* Hint: You should be able to knit the script if all code is working correctly...
```
Comment (vincenzocoia) [inline 2020-materials/team-troubleshooting/TB1.Rmd:25]: Nice!

File: 2020-materials/team-troubleshooting/student-instructions.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+### Learning Objectives
+
+1. Collaborate on a project with a partner using GitHub
+2. Troubleshoot broken code
+3. Rewrite code to address instructions
+
+### Workflow
+
+1. Create a group repository.
```
[the remark this answers is not in the document]
Comment (sciclic) [reply 2020-materials/team-troubleshooting/student-instructions.md:11]: Sounds great! Thank you.

File: 2020-materials/team-troubleshooting/student-instructions.md (hunk 1 above)
Comment (sciclic) [reply 2020-materials/team-troubleshooting/student-instructions.md:40]: Yup, that was the idea. Completing would be 10 points, and then adjustments can be made if there is any problem with the teamwork. Having a pre-determined multiplier sounds good!

**Comment by sciclic** [comment]
Regarding the grading, I agree that having two documents may be confusing. What I did to generate the first Rmd for the team troubleshooting doc was:

1. Writing correct code
2. Copy-paste the full Rmd into a new file
3. Create broken code or errors (code that does not follow instructions)
3. Alternate between the two to mark ### SOLUTION HERE ### every time I introduce an error (either an unmarked error, or an ### ERROR HERE ###)

Indeed this allows for more "errors" or inconsistencies between the two documents to occur. There is the [learnR](https://github.com/rstudio/learnr) package that allows to turn any R Markdown document into an interactive tutorial, and can be combined with [gradethis](https://github.com/rstudio-education/gradethis), but I think that the output is an online Shiny document and I'm not sure of how they would submit their work. It actually looks quite interesting (could actually be an alternative to nbgrader for the lecture worksheets in future years) and there must be a way to do so, but given the timing I'm not sure that there will be time to get familiar with the workflow for this purpose.

nbgrader seems like a reasonably straightforward solution, but at the same time I am concerned that then the students' only interaction with the R Studio interface will be during their mini data-analysis project, since the lecture worksheets will be completed as a Jupyter notebook for the purpose of nbgrader. In summary, I think that either using nbgrader or leaving it as is for this year and attempt to find a better solution are good options; please let me know your thoughts!

In any case, I will attempt to complete this team troubleshooting worksheet 1 today.

Thank you,
Icíar

**Comment by vincenzocoia** [comment]
Let's stick with 2 separate documents for now, then.

**Pull request merged**